bindings/cpp/AndroidSecureStorage/app/src/main/java/com/cronos/play/SecureStorage.kt

```diff
@@ -0,0 +1,71 @@
+package com.cronos.play
+
+import android.content.Context
+import androidx.security.crypto.EncryptedFile
+import androidx.security.crypto.MasterKey
+import androidx.security.crypto.MasterKey.Builder
+import java.io.File
+
+class SecureStorage {
+
+  companion object {
+    init {}
+
+    // read from secure storage
+    // return json string
+    // success: {"result":"value","success":"1","error":""}
+    // fail: {"result":"","success":"0","error":"encrypt file not found"}
+    @JvmStatic
+    fun readSecureStorage(context: Context, key: String): String {
+
+      val masterKey: MasterKey =
+          Builder(context).setKeyScheme(MasterKey.KeyScheme.AES256_GCM).build()
+
+      val file = File(context.filesDir, key)
+      val encryptedFile: EncryptedFile =
+          EncryptedFile.Builder(
+                  context, file, masterKey, EncryptedFile.FileEncryptionScheme.AES256_GCM_HKDF_4KB)
+              .build()
+
+      var ret: String
+      if (file.exists()) {
+        encryptedFile.openFileInput().use { inputStream ->
+          var myvalue = inputStream.readBytes().toString(Charsets.UTF_8)
+          ret = String.format("{\"result\":\"%s\",\"success\":\"1\",\"error\":\"\"}", myvalue)
+        }
+      } else {
+        ret =
+            String.format(
+                "{\"result\":\"\",\"success\":\"0\",\"error\":\"encrypt file not found\"}")
+      }
```

Comment:
seems ok, but you can perhaps run `ktlint` (https://pinterest.github.io/ktlint/) for suggestions to make it more idiomatic. e.g. I think one could do. `val ret = if (...) { ... }` and Kotlin has string inteporlation (https://kotlincompact.com/string-interpolation.html), so no need to call `String.format` (it should be possible just to write `"{\"result\":\"$myvalue\",..`)

Reply:
ok